sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -359,6 +359,16 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       return df
     }
 
+    val replaceableAttrs = AttributeSet(cols.map { colName =>
+      // Check column name exists
+      val attr = df.resolve(colName) match {
+        case a: Attribute => a
+        case _ => throw new UnsupportedOperationException(
+          s"Nested field ${colName} is not supported.")
```

Comment:
The current behaviour just ignores input if given columns do not exist?
```
scala> df.na.replace("xxx", Map("n/a" -> "Unknown")).show()
+---+---+
| c0| c1|
+---+---+
|abc| 23|
|def| 44|
|n/a|  0|
+---+---+
```

Reply:
`df.resolve` throws `org.apache.spark.sql.AnalysisException` with error message `Cannot resolve column name "xxx" among (c0, c1)`, if any of the given columns do not exist.

Reply:
Could you add tests for these exceptions?

Reply:
Added.